### DISCUSSION TITLE
Support for OIDC AuthProvider in RefineV4

### DISCUSSION THREAD

**Discussion by sunilranaLivspace** [refinedev/refine#6477] [Q&A] [answered]
Can someone help me with OIDC authentication integration with Refine V4?
Someone has done it for RefineV3 - https://github.com/cloudstorydev/refine-oidc-provider
But I do the similar changes in Refine v4, it is not working properly and breaking.

**Comment by BatuhanW** [comment, marked as answer]
Hello @sunilranaLivspace implementation shouldn't be much different. In refine auth provider's login method, you can call your OIDC provider redirect method. Then it will redirect you to a path you defined. You can create a page here, for example `my-app/redirect` and parse tokens in the query params, and persist them in your storage. 

Only thing you need to do is to implement `getIdentity` method, which will fetch token and/or user information from your oidc provider storage and return user. You can also implement `onError` method to handle errors and redirect user back to login page.
